canard.c

```diff
@@ -434,21 +434,18 @@ int16_t canardHandleRxFrame(CanardInstance* ins, const CanardCANFrame* frame, ui
     uint64_t data_type_signature = 0;
     CanardRxState* rx_state = NULL;
 
+    if (!ins->should_accept(ins, &data_type_signature, data_type_id, transfer_type, source_node_id))
```

Comment:
should_accept() can be quite an expensive function on some systems, for example in the C++ API it traverses a linked list which can be quite long. I agree with the semantic change that we should return CANARD_ERROR_RX_NOT_WANTED, but I think it would be better to do it after the findRxState() call in the !IS_START_OF_TRANSFER() part of the if(). In other words, we call ins->should_accept() in order to decide if we should return CANARD_ERROR_RX_NOT_WANTED or CANARD_ERROR_RX_MISSED_START in the error path. That way in normal operation we would only be calling should_accept() once per message